From c5399667c11015d58e2869344c3e73065e67b073 Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Mon, 12 Oct 2020 16:35:23 +0300
Subject: [PATCH 1/5] arm64: dts: renesas: r8a77970: Add ES1 and Eagle Function
 boards

This backports r8a77970-es1 Eagle Function boards from BSP7.
The UIO support is dropped for now.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/Makefile               |   4 +
 .../boot/dts/renesas/r8a77970-eagle-function.dts   |  62 +++++++++++
 .../dts/renesas/r8a77970-es1-eagle-function.dts    |  17 +++
 arch/arm64/boot/dts/renesas/r8a77970-es1-eagle.dts |  17 +++
 arch/arm64/boot/dts/renesas/r8a77970-es1-v3msk.dts |  17 +++
 arch/arm64/boot/dts/renesas/r8a77970-es1.dtsi      | 116 +++++++++++++++++++++
 6 files changed, 233 insertions(+)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a77970-eagle-function.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a77970-es1-eagle-function.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a77970-es1-eagle.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a77970-es1-v3msk.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a77970-es1.dtsi

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index cbd7a7a..a12267e 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -26,6 +26,10 @@ dtb-$(CONFIG_ARCH_R8A77990) += r8a77990-ebisu-4d.dtb
 dtb-$(CONFIG_ARCH_R8A77990) += r8a77990-es10-ebisu.dtb r8a77990-es10-ebisu-4d.dtb
 dtb-$(CONFIG_ARCH_R8A77995) += r8a77995-draak.dtb
 
+dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-eagle-function.dtb
+dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-es1-eagle.dtb r8a77970-es1-eagle-function.dtb
+dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-es1-v3msk.dtb
+
 # overlays
 DTC_FLAGS += -@
 ulcb-kf-dtbo := ulcb-kf-cn11.dtbo ulcb-kf-most.dtbo ulcb-kf-sd3.dtbo \
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-eagle-function.dts b/arch/arm64/boot/dts/renesas/r8a77970-eagle-function.dts
new file mode 100644
index 0000000..dab17bf
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a77970-eagle-function.dts
@@ -0,0 +1,62 @@
+/*
+ * Device Tree Source for the Eagle Function board on r8a7797
+ *
+ * Copyright (C) 2017 Renesas Electronics Corp.
+ * Copyright (C) 2017 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+#include "r8a77970-eagle.dts"
+
+/ {
+	model = "Renesas Eagle Function board based on r8a7797";
+
+	vcc_3v3: regulator0 {
+		compatible = "regulator-fixed";
+		regulator-name = "fixed-VCC3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	vcc_vddq_vin0: regulator1 {
+		compatible = "regulator-fixed";
+		regulator-name = "VCC-VDDQ-VIN0";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+};
+
+&pfc {
+	sdhi2_pins_1v8: sdhi2_1v8 {
+		groups = "mmc_data8", "mmc_ctrl";
+		function = "mmc";
+		power-source = <1800>;
+	};
+
+	sdhi2_pins_3v3: sdhi2_3v3 {
+		groups = "mmc_data8", "mmc_ctrl";
+		function = "mmc";
+		power-source = <3300>;
+	};
+};
+
+&mmc0 {
+	/* used for on-board eMMC */
+	pinctrl-0 = <&sdhi2_pins_3v3>;
+	pinctrl-1 = <&sdhi2_pins_1v8>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&vcc_3v3>;
+	vqmmc-supply = <&vcc_vddq_vin0>;
+	mmc-hs200-1_8v;
+	bus-width = <8>;
+	non-removable;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle-function.dts b/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle-function.dts
new file mode 100644
index 0000000..b7e74f3
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle-function.dts
@@ -0,0 +1,17 @@
+/*
+ * Device Tree Source for the Eagle Function board on r8a7797 ES1.0
+ *
+ * Copyright (C) 2018 Renesas Electronics Corp.
+ * Copyright (C) 2018 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+#include "r8a77970-eagle-function.dts"
+#include "r8a77970-es1.dtsi"
+
+/ {
+	model = "Renesas Eagle Function board based on r8a7797 ES1.0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle.dts b/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle.dts
new file mode 100644
index 0000000..13e0a19
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a77970-es1-eagle.dts
@@ -0,0 +1,17 @@
+/*
+ * Device Tree Source for the Eagle board on r8a7797 ES1.0
+ *
+ * Copyright (C) 2018 Renesas Electronics Corp.
+ * Copyright (C) 2018 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+#include "r8a77970-eagle.dts"
+#include "r8a77970-es1.dtsi"
+
+/ {
+	model = "Renesas Eagle board based on r8a7797 ES1.0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-es1-v3msk.dts b/arch/arm64/boot/dts/renesas/r8a77970-es1-v3msk.dts
new file mode 100644
index 0000000..1d955f3
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a77970-es1-v3msk.dts
@@ -0,0 +1,17 @@
+/*
+ * Device Tree Source for the V3M Starter Kit board on r8a7797 ES1.0
+ *
+ * Copyright (C) 2018 Renesas Electronics Corp.
+ * Copyright (C) 2018 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+#include "r8a77970-v3msk.dts"
+#include "r8a77970-es1.dtsi"
+
+/ {
+	model = "Renesas V3M Starter Kit board based on r8a7797 ES1.0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-es1.dtsi b/arch/arm64/boot/dts/renesas/r8a77970-es1.dtsi
new file mode 100644
index 0000000..ca004a6
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a77970-es1.dtsi
@@ -0,0 +1,116 @@
+/*
+ * Device Tree Source for the r8a7797 SoC ES1.0 SoC
+ *   (append to r8a7797 SoC ES2.0 SoC)
+ *
+ * Copyright (C) 2018 Renesas Electronics Corp.
+ * Copyright (C) 2018 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+/ {
+	soc {
+		imp_distributer: imp_ctrl_00 {
+			compatible = "renesas,impx4-distributer";
+			reg = <0 0xffa00000 0 0x10000>;
+			interrupts = <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 830>;
+			power-domains = <&sysc R8A77970_PD_A3IR>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+		};
+
+		imp_imp_00 {
+			compatible = "renesas,impx4-legacy";
+			reg = <0 0xff900000 0 0x20000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <0>;
+			clocks = <&cpg CPG_MOD 827>;
+			power-domains = <&sysc R8A77970_PD_A2IR0>;
+		};
+
+		imp_imp_01 {
+			compatible = "renesas,impx4-legacy";
+			reg = <0 0xff920000 0 0x20000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <1>;
+			clocks = <&cpg CPG_MOD 826>;
+			power-domains = <&sysc R8A77970_PD_A2IR1>;
+		};
+
+		imp_imp_02 {
+			compatible = "renesas,impx4-legacy";
+			reg = <0 0xff940000 0 0x20000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <2>;
+			clocks = <&cpg CPG_MOD 825>;
+			power-domains = <&sysc R8A77970_PD_A2DP>;
+		};
+
+		imp_imp_03 {
+			compatible = "renesas,impx4-legacy";
+			reg = <0 0xff960000 0 0x20000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <3>;
+			clocks = <&cpg CPG_MOD 824>;
+			power-domains = <&sysc R8A77970_PD_A2CN>;
+		};
+
+		imp_ocv_00 {
+			compatible = "renesas,impx4-shader";
+			reg = <0 0xff980000 0 0x10000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <4>;
+			clocks = <&cpg CPG_MOD 829>;
+			power-domains = <&sysc R8A77970_PD_A2SC0>;
+		};
+
+		imp_ocv_01 {
+			compatible = "renesas,impx4-shader";
+			reg = <0 0xff990000 0 0x10000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <5>;
+			clocks = <&cpg CPG_MOD 828>;
+			power-domains = <&sysc R8A77970_PD_A2SC1>;
+		};
+
+		imp_dma_00 {
+			compatible = "renesas,impx5-dmac";
+			reg = <0 0xffa10000 0 0x1000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <16>;
+			clocks = <&cpg CPG_MOD 830>;
+			power-domains = <&sysc R8A77970_PD_A3IR>;
+		};
+
+		imp_dma_01 {
+			compatible = "renesas,impx5-dmac";
+			reg = <0 0xffa10000 0 0x1000>,
+			      <0 0xffa10800 0 0x0800>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <17>;
+			clocks = <&cpg CPG_MOD 830>;
+			power-domains = <&sysc R8A77970_PD_A3IR>;
+		};
+
+		imp_psc_00 {
+			compatible = "renesas,impx5+-psc";
+			reg = <0 0xffa20000 0 0x4000>;
+			interrupt-parent = <&imp_distributer>;
+			interrupts = <12>;
+			clocks = <&cpg CPG_MOD 830>;
+			power-domains = <&sysc R8A77970_PD_A3IR>;
+		};
+
+		/delete-node/imp_cnn_00;
+
+		imp_mem_00 {
+			compatible = "renesas,impx4-memory";
+			reg = <0 0xed000000 0 0xe0000>;
+			clocks = <&cpg CPG_MOD 830>;
+			power-domains = <&sysc R8A77970_PD_A3IR>;
+		};
+	};
+};
-- 
2.7.4

